README.md
# t2
[![Build Status](https://travis-ci.org/cdcabrera/t2.svg?branch=master)](https://travis-ci.org/cdcabrera/apidoc-mock)
[![codecov](https://codecov.io/gh/cdcabrera/t2/branch/master/graph/badge.svg)](https://codecov.io/gh/cdcabrera/apidoc-mock)
[![License](https://img.shields.io/github/license/cdcabrera/t2.svg)](https://github.com/cdcabrera/apidoc-mock/blob/master/LICENSE)

testing more build

## Requirements
The basic requirements:
 * [NodeJS version 10+](https://nodejs.org/)
 * Optionally your system could be running
    - [Yarn 1.16+](https://yarnpkg.com), otherwise NPM should be adequate.
    - [Docker](https://docs.docker.com/engine/installation/)
    
## Test
Test change again again again again


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.2.5](https://github.com/cdcabrera/t2/compare/v0.2.4...v0.2.5) (2020-05-07)


### Bug Fixes

* **add push:** stage ([272438e](https://github.com/cdcabrera/t2/commit/272438e))
* **build:** master and stage workflows ([646fbd9](https://github.com/cdcabrera/t2/commit/646fbd9))
* **build:** pr setup ([a351e2f](https://github.com/cdcabrera/t2/commit/a351e2f))
* **fetch:** stage workflow ([538d5f2](https://github.com/cdcabrera/t2/commit/538d5f2))
* **frust:** ci ([2e153bf](https://github.com/cdcabrera/t2/commit/2e153bf))
* **it:** now ([7c9e4ec](https://github.com/cdcabrera/t2/commit/7c9e4ec))
* **pr:** body ([57561fc](https://github.com/cdcabrera/t2/commit/57561fc))
* **stage:** checkout ([8b051ee](https://github.com/cdcabrera/t2/commit/8b051ee))
* **stage:** multiline pr body ([400f7a5](https://github.com/cdcabrera/t2/commit/400f7a5))
* **stage:** pr body fix ([1949083](https://github.com/cdcabrera/t2/commit/1949083))
* **stage:** step name ([0330c7c](https://github.com/cdcabrera/t2/commit/0330c7c))
* **stage:** workflow ([f848295](https://github.com/cdcabrera/t2/commit/f848295))
* **stage:** workflow ([092f073](https://github.com/cdcabrera/t2/commit/092f073))
* **stage:** workflow ([72b1573](https://github.com/cdcabrera/t2/commit/72b1573))
* **stage:** workflow job title ([215ee09](https://github.com/cdcabrera/t2/commit/215ee09))
* **stage:** workflow job title ([81e239d](https://github.com/cdcabrera/t2/commit/81e239d))
* **stage:** workflow pr title check, node test ([1df6149](https://github.com/cdcabrera/t2/commit/1df6149))



### [0.2.4](https://github.com/cdcabrera/t2/compare/v0.2.3...v0.2.4) (2020-05-07)


### Bug Fixes

* **docs:** readme updated ([5785a1c](https://github.com/cdcabrera/t2/commit/5785a1c))



### [0.2.3](https://github.com/cdcabrera/t2/compare/v0.2.2...v0.2.3) (2020-05-07)


### Bug Fixes

* **build:** master workflow ([1bafb5b](https://github.com/cdcabrera/t2/commit/1bafb5b))



### [0.2.2](https://github.com/cdcabrera/t2/compare/v0.2.1...v0.2.2) (2020-05-07)


### Bug Fixes

* **check:** master ([2d22c90](https://github.com/cdcabrera/t2/commit/2d22c90))



### [0.2.1](https://github.com/cdcabrera/t2/compare/v0.2.0...v0.2.1) (2020-05-07)



## [0.2.0](https://github.com/cdcabrera/t2/compare/v0.1.3...v0.2.0) (2020-05-07)


### Features

* **readme:** docs ([e7417d9](https://github.com/cdcabrera/t2/commit/e7417d9))



### [0.1.3](https://github.com/cdcabrera/t2/compare/v0.1.2...v0.1.3) (2020-05-06)



### [0.1.2](https://github.com/cdcabrera/t2/compare/v0.1.1...v0.1.2) (2020-05-06)


### Bug Fixes

* **build:** clean up ([d6a4851](https://github.com/cdcabrera/t2/commit/d6a4851))



### [0.1.1](https://github.com/cdcabrera/t2/compare/v0.1.0...v0.1.1) (2020-05-06)


### Bug Fixes

* **build:** release ([7452d87](https://github.com/cdcabrera/t2/commit/7452d87))
* **build:** release ([b433d09](https://github.com/cdcabrera/t2/commit/b433d09))
* **check:** user ([1dcefe4](https://github.com/cdcabrera/t2/commit/1dcefe4))
* **workflow:** test ([ea82e2c](https://github.com/cdcabrera/t2/commit/ea82e2c))
* **workflow:** test ([eba2cae](https://github.com/cdcabrera/t2/commit/eba2cae))



## 0.1.0 (2020-05-05)


### Features

* **build:** nodejs action ([ec92c7b](https://github.com/cdcabrera/t2/commit/ec92c7b82ff0c2fcc328b67850dbc8881992dcdd))
* **build:** setup repo ([e451c00](https://github.com/cdcabrera/t2/commit/e451c0022fd816e0760e1b56862616ee69c30eea))


### Bug Fixes

* **again:** why ([0b8c10a](https://github.com/cdcabrera/t2/commit/0b8c10a1228c730fd947d1be9a1d786ea96b9b70))
* **build:** again ([de313ff](https://github.com/cdcabrera/t2/commit/de313ff07f01e6146e8bd589b12a80882a8b131e))
* **build:** apply workflow to ci and qa branches ([3bcc70e](https://github.com/cdcabrera/t2/commit/3bcc70e0145c39a03314f00745da998edb023a87))
* **build:** ci ([157bf7f](https://github.com/cdcabrera/t2/commit/157bf7f91465bae88a89e1fcfc1a0b81d273f355))
* **build:** ci ([04e4381](https://github.com/cdcabrera/t2/commit/04e4381dac7aefe70bc3ebb6d7648fa0b8f34a86))
* **build:** ci ([9e651e7](https://github.com/cdcabrera/t2/commit/9e651e7f874af9674424b3ea04f962c4e61f8b6a))
* **build:** ci ([8986941](https://github.com/cdcabrera/t2/commit/898694187dea91ab033f26fd6bcd0fdd235f3558))
* **build:** ci again ([05ca5f7](https://github.com/cdcabrera/t2/commit/05ca5f7a064f6a7ddd4c5afa9f2a80f999050fe6))
* **build:** ci check ([6804862](https://github.com/cdcabrera/t2/commit/68048627ab0fa771659180f36f63bf449142030d))
* **build:** packagejson ([534bfc5](https://github.com/cdcabrera/t2/commit/534bfc5f330ba22c21c16e250198d45ee44417a4))
* **readme:** copy again ([57dfd00](https://github.com/cdcabrera/t2/commit/57dfd00291ad21850473801545f0fc1ee6e0ef01))
* **readme:** copy update ([b024297](https://github.com/cdcabrera/t2/commit/b024297c44b36e108ee8f0dfbc112bd9df8a36a5))
* **testing:** workflow ([fc98e54](https://github.com/cdcabrera/t2/commit/fc98e544dd7d36e7ce106cddbe536258c3d0cdc9))

# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.


src/__tests__/index.test.js
const { t2, opts } = require('../');

describe('t2', () => {
  it('should have specific defined methods and properties', () => {
    expect(t2).toBeDefined();
    expect(opts).toBeDefined();
  });

  it('should export an object', () => {
    expect(t2('testing')).toMatchSnapshot('output');
  });
});


scripts/build.sh
#!/usr/bin/env bash
#
#
# Build
#
build()
{
  local SRC=$1
  local DIR=$2
  mkdir -p $DIR
  printf "${YELLOW}Building ...${NOCOLOR}"
  cp $SRC/* $DIR 2>/dev/null
  printf "${GREEN}SUCCESS${NOCOLOR}\n"
}
#
#
# Git commit hash and package.json version
#
version()
{
  local DIR=$1

  UI_VERSION="$(node -p 'require(`./package.json`).version').$(git rev-parse --short HEAD)"
  printf "Version ${YELLOW}UI_VERSION=$UI_VERSION${NOCOLOR} ...${NOCOLOR}"
  mkdir -p $DIR
  echo UI_VERSION="$UI_VERSION" >> $DIR/version.txt
  printf "${GREEN}SUCCESS${NOCOLOR}\n"
}
#
#
# Clean directories
#
clean()
{
  local DIR=$1

  echo "Cleaning build directories, files..."
  rm -rf -- $DIR
}
#
#
# main()
#
{
  BLUE="\e[34m"
  RED="\e[31m"
  GREEN="\e[32m"
  YELLOW="\e[33m"
  NOCOLOR="\e[39m"

  SRC_DIR=./src
  BUILD_DIR=./build

  clean $BUILD_DIR
  version $BUILD_DIR
  build $SRC_DIR $BUILD_DIR
}


.travis/rebase.sh
#!/bin/bash
#
#
# Clone, build for local development
#
gitRebase()
{
    local GITREMOTE=$1
    local GITBRANCH=$2
    local GITSOURCEBRANCH=$3

    printf "${YELLOW}Attempting to rebase branch \"${GITBRANCH}\" ...${NOCOLOR}"

    if [ ! -z "$(git branch -a | grep $GITBRANCH)" ]; then
      git checkout ${GITBRANCH}
      git rebase ${GITSOURCEBRANCH}
      git push -f ${GITREMOTE} ${GITBRANCH} --quiet
      printf "${GREEN}SUCCESS${NOCOLOR}\n"
    else
      printf "${YELLOW}SKIPPING${NOCOLOR}\n"
    fi
}
#
#
# main()
#
{
  set -e
  set -x

  GREEN="\e[32m"
  YELLOW="\e[33m"
  NOCOLOR="\e[39m"

  # filter
  #if [ "${AUTO_REBASE}" != "true" ] || [ "${TRAVIS_PULL_REQUEST}" != "false" ] || [[ $REPO != *"$TRAVIS_REPO_SLUG"* ]]; then
  if [ -z "${AUTO_REBASE}" ] || [ "${TRAVIS_PULL_REQUEST}" != "false" ] || [[ $REPO != *"$TRAVIS_REPO_SLUG"* ]]; then
    echo -e "${YELLOW}Exiting early, rebase conditions not met.${NOCOLOR}"
    exit 0;
  fi

  # filter process for specific branch and CI stage
  if [[ "${TRAVIS_BRANCH}" = "${BRANCH}" ]] && [[ $TRAVIS_BUILD_STAGE_NAME == *"Rebase"* ]]; then
    set +x
    openssl aes-256-cbc \
            -K `env | grep 'encrypted_2.*_key' | cut -f2 -d '='` \
            -iv `env | grep 'encrypted_2.*_iv' | cut -f2 -d '='` \
            -in .travis/release_key.enc -out .travis/release_key -d
    set -x

    chmod 600 .travis/release_key
    eval `ssh-agent -s`
    ssh-add .travis/release_key

    echo -e "${YELLOW}Attempting rebase...${NOCOLOR}\n"

    git config --global user.name ${USER}
    git config --global user.email ${EMAIL}
    git remote add ssh-origin ${REPO}
    git checkout ${BRANCH}
    #git fetch ssh-origin ${BRANCH}
    #git reset --hard ssh-origin/${BRANCH}

    for RB_BRANCH in ${AUTO_REBASE//,/ }
    do
       gitRebase $REPO $RB_BRANCH $BRANCH
    done

    #gitRebase $REPO "stage" $BRANCH
    #gitRebase $REPO "qa" $BRANCH
    #gitRebase $REPO "ci" $BRANCH

    echo -e "${GREEN}Rebase completed.${NOCOLOR}"
    exit 0;
  fi

  # fallback messaging
  echo -e "${YELLOW}Exiting, nothing to rebase${NOCOLOR}"
  exit 0;
}

}


.travis/release.sh
#!/bin/bash
#
#
# main()
#
{
  set -e
  set -x

  GREEN="\e[32m"
  YELLOW="\e[33m"
  NOCOLOR="\e[39m"
  #DIFF_CHECK=$(git diff | grep '^+' | grep -v '+++' | wc -l)

  # filter out pull requests
  if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then
    echo -e "${YELLOW}Exiting early, pull request${NOCOLOR}"
    exit 0;
  fi

  # filter out forked repositories
  if [[ $REPO != *"$TRAVIS_REPO_SLUG"* ]]; then
    echo -e "${YELLOW}Exiting early, not master repository${NOCOLOR}"
    exit 0;
  fi

  # if auto deploy active filter out timestamp only updates
  #if [ "${AUTO_DEPLOY}" = "true" ] && [[ "${DIFF_CHECK}" = "0" ]]; then
  #  echo -e "${YELLOW}$(git diff --compact-summary){$NOCOLOR}"
  #  echo -e "${YELLOW}Exiting early, no significant changes${NOCOLOR}"
  #  exit 0;
  #fi

  # filter out deployment tokens when active
  if [ "${AUTO_DEPLOY}" != "true" ] && [[ ! "$TRAVIS_COMMIT_MESSAGE" =~ "[RELEASE]" ]]; then
    echo -e "${YELLOW}Exiting early, not a release${NOCOLOR}"
    exit 0;
  fi

  # filter process for specific branch and CI stage
  #if [[ "${TRAVIS_BRANCH}" = "${BRANCH}" ]] && [[ $TRAVIS_BUILD_STAGE_NAME == *"Build"* ]]; then
  if [[ "${TRAVIS_BRANCH}" = "${BRANCH}" ]]; then
    set +x
    openssl aes-256-cbc \
            -K `env | grep 'encrypted_2.*_key' | cut -f2 -d '='` \
            -iv `env | grep 'encrypted_2.*_iv' | cut -f2 -d '='` \
            -in .travis/release_key.enc -out .travis/release_key -d
    set -x

    chmod 600 .travis/release_key
    eval `ssh-agent -s`
    ssh-add .travis/release_key

    git config --global user.name ${USER}
    git config --global user.email ${EMAIL}
    git remote add ssh-origin ${REPO}

    echo -e "${YELLOW}PUSHING release...${NOCOLOR}"

    #cat package.json
    ls
    #git fetch ssh-origin ${BRANCH}
    git fetch --tags
    git checkout ${BRANCH}
    #standard-version --dry-run
    yarn release --dry-run

    wait
    git push --follow-tags ssh-origin ${BRANCH}

    echo -e "${GREEN}COMPLETED release${NOCOLOR}"
    exit 0;
  fi

  # fallback messaging
  echo -e "${YELLOW}Exiting, not ${BRANCH} branch or build stage${NOCOLOR}"
  exit 0;
}
